### PR TITLE
Update stockfish to 16.1

### PR DESCRIPTION
**Pull request by veloce**
The example project runs fine on android, but not on iOS:

```
flutter: The Dart VM service is listening on http://127.0.0.1:63359/S9GONOBwFL0=/
flutter: FINE: 2024-04-12 11:45:37.106271: The init isolate reported StockfishState.ready
flutter: FINEST: 2024-04-12 11:45:37.110244: The stdout isolate sent Stockfish 16.1 by the Stockfish developers (see AUTHORS file)
flutter: FINEST: 2024-04-12 11:45:39.298636: The stdout isolate sent readyok
flutter: FINEST: 2024-04-12 11:45:46.607387: The stdout isolate sent info string ERROR: Network evaluation parameters compatible with the engine must be available.
flutter: FINEST: 2024-04-12 11:45:46.608057: The stdout isolate sent info string ERROR: The network file nn-b1a57edbea57.nnue was not loaded successfully.
flutter: FINEST: 2024-04-12 11:45:46.608311: The stdout isolate sent info string ERROR: The UCI option EvalFile might need to specify the full path, including the directory name, to the network file.
flutter: FINEST: 2024-04-12 11:45:46.608528: The stdout isolate sent info string ERROR: The default net can be downloaded from: https://tests.stockfishchess.org/api/nn/nn-b1a57edbea57.nnue
flutter: FINEST: 2024-04-12 11:45:46.608648: The stdout isolate sent info string ERROR: The engine will be terminated now.
```

It somehow cannot load NNUE file. Is it a path issue?

TODO:

- [x] fix iOS example

### CODE REVIEW

**Comment by daohoangson** [comment]
Did iOS work?

**Comment by veloce** [comment]
No it still doesn't work in the example app. 
The engine loads fine, but not the network file. 
I have no idea why and it's weird because it should be embedded in the binary during compilation.
I guess we need to debug it.

**Comment by daohoangson** [comment]
Did you test with real device? Which iOS version? I have just run a quick check on 17.4 simulator (M1 laptop) and it seemed to work fine.

**Comment by veloce** [comment]
No I only tested with a simulator also on a M1 laptop. 

Will try a real device and other simulators.

**Comment by daohoangson** [comment]
That's weird...

**Comment by veloce** [comment]
Your simulator version is 17.4? Mine is 15.1... I'll try to update but I doubt this is the cause.

**Comment by veloce** [comment]
So I've updated my simulator now it's using iOS 17.4, and restarted my computer. Now the example is working! I've no idea why it was not working before.

**Comment by veloce** [comment]
I think we could further optimise iOS with the flags `-DUSE_NEON=8` and even `-DUSE_NEON_DOTPROD`.

The problem is that it doesn't work if we run the app in debug mode. I got this error, running the example app on my M1 laptop, with the simulator or a real device:
```
User-Defined Issue (Xcode): "NEON intrinsics not available with the soft-float
ABI. Please use -mfloat-abi=softfp or -mfloat-abi=hard"
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/
usr/lib/clang/15.0.0/include/arm_neon.h:27:1
```

I can build without problem in release mode though, and if I install the IPA it works fine.

@daohoangson do you have an idea on how we could solve this problem? Maybe we can define flags only in release mode, but I don't know how to do that.

**Comment by daohoangson** [comment]
https://pewpewthespells.com/blog/xcconfig_guide.html#ConditionalVariableAssignment and https://stackoverflow.com/a/34200599/308114 look promising. Maybe something like these:

```
  s.xcconfig = {
    'OTHER_CPLUSPLUSFLAGS[config=Debug]' => '$(inherited) -blah-blah',
  }
```

I think we should investigate why it fails the debug build though.

**Comment by veloce** [comment]
I tested with the different flags for debug and release and it works well @daohoangson 

There is no need to investigate why neon support is failing in debug, because it does not make sense to activate it in debug mode. People using x86_64 computers should be able to run the project in their simulators.

File: ios/stockfish.podspec [hunk 1]
```diff
@@ -19,23 +19,32 @@ Pod::Spec.new do |s|
   s.public_header_files = 'Classes/**/*.h'
   s.exclude_files = 'Stockfish/src/incbin/UNLICENCE'
   s.dependency 'Flutter'
-  s.platform = :ios, '11.0'
-  s.ios.deployment_target  = '11.0'
+  s.platform = :ios, '12.0'
+  s.ios.deployment_target  = '12.0'
```
Comment (daohoangson) [inline ios/stockfish.podspec:23]: Do we require `IPHONEOS_DEPLOYMENT_TARGET>=12` from now on?

Comment (veloce) [reply ios/stockfish.podspec:23]: I just followed Flutter which now supports 12+

Comment (daohoangson) [reply ios/stockfish.podspec:23]: Let's bump iOS version in README then.

Comment (veloce) [reply ios/stockfish.podspec:23]: 👍  I just updated the README

**Review by daohoangson — APPROVED** [review]

**Comment by Mercutio1243** [comment]
Update seems to be stable. But can it be that the package size has increase significantly? How can I find this out?

**Comment by daohoangson** [comment]
The app size is expected to increase with Stockfish 16.1 because of the two NNUE files (total size of 60MB). That's why @veloce has created another PR to revert back to Stockfish 16 (1 file of 40MB) to keep the app size smaller, see #41.

**Pull request merged**